tests/tidy.rs

```diff
@@ -0,0 +1,46 @@
+use std::{
+    env,
+    path::{Path, PathBuf},
+    process::{Command, Stdio},
+};
+
+fn project_root() -> PathBuf {
+    PathBuf::from(
+        env::var("CARGO_MANIFEST_DIR").unwrap_or_else(|_| env!("CARGO_MANIFEST_DIR").to_owned()),
+    )
+}
+
+fn run(cmd: &str, dir: impl AsRef<Path>) -> Result<(), ()> {
```

Comment:
For non-public APIs, I prefer to avoid `AsRef` polymorphism, to save on compile times. But this probably doesn't matter. 

bors r+